soroban-env-host/src/host/declared_size.rs

```diff
@@ -0,0 +1,438 @@
+        expect!["77"].assert_eq(size_of::<LedgerKeyTrustLine>().to_string().as_str());
+        expect!["40"].assert_eq(size_of::<LedgerKeyOffer>().to_string().as_str());
+        expect!["56"].assert_eq(size_of::<LedgerKeyData>().to_string().as_str());
+        expect!["32"].assert_eq(size_of::<LedgerKeyClaimableBalance>().to_string().as_str());
+        expect!["32"].assert_eq(size_of::<LedgerKeyLiquidityPool>().to_string().as_str());
+        expect!["32"].assert_eq(size_of::<LedgerKeyContractCode>().to_string().as_str());
+        expect!["4"].assert_eq(size_of::<LedgerKeyConfigSetting>().to_string().as_str());
+        expect!["33"].assert_eq(size_of::<LedgerEntryExt>().to_string().as_str());
+        expect!["216"].assert_eq(size_of::<AccountEntry>().to_string().as_str());
+        expect!["128"].assert_eq(size_of::<TrustLineEntry>().to_string().as_str());
+        expect!["152"].assert_eq(size_of::<OfferEntry>().to_string().as_str());
+        expect!["80"].assert_eq(size_of::<DataEntry>().to_string().as_str());
+        expect!["120"].assert_eq(size_of::<ClaimableBalanceEntry>().to_string().as_str());
+        expect!["160"].assert_eq(size_of::<LiquidityPoolEntry>().to_string().as_str());
+        expect!["56"].assert_eq(size_of::<ContractCodeEntry>().to_string().as_str());
+        expect!["8"].assert_eq(size_of::<ConfigSettingEntry>().to_string().as_str());
+        expect!["1"].assert_eq(size_of::<AccessType>().to_string().as_str());
+        expect!["16"].assert_eq(size_of::<DebugArg>().to_string().as_str());
+        expect!["40"].assert_eq(size_of::<InternalContractEvent>().to_string().as_str());
+        expect!["80"].assert_eq(size_of::<DebugEvent>().to_string().as_str());
+        expect!["104"].assert_eq(size_of::<ContractEvent>().to_string().as_str());
+        expect!["112"].assert_eq(size_of::<HostEvent>().to_string().as_str());
+        expect!["24"].assert_eq(size_of::<Events>().to_string().as_str());
+        expect!["80"].assert_eq(size_of::<InternalEvent>().to_string().as_str());
+        expect!["24"].assert_eq(size_of::<ScBytes>().to_string().as_str());
+        expect!["24"].assert_eq(size_of::<ScString>().to_string().as_str());
+        expect!["24"].assert_eq(size_of::<ScSymbol>().to_string().as_str());
+        // composite types
+        expect!["16"].assert_eq(size_of::<&[ScVal]>().to_string().as_str());
+        expect!["48"].assert_eq(size_of::<(RawVal, ScVal)>().to_string().as_str());
+        expect!["200"].assert_eq(size_of::<[ScVal; 5]>().to_string().as_str());
+        expect!["24"].assert_eq(size_of::<BytesM<10000>>().to_string().as_str());
+        expect!["24"].assert_eq(size_of::<StringM<10000>>().to_string().as_str());
+        expect!["24"].assert_eq(size_of::<Vec<ScVal>>().to_string().as_str());
+        expect!["8"].assert_eq(size_of::<Box<ScVal>>().to_string().as_str());
+        // For these below we don't reply on the size_of, since it is the lower bound
```

Comment:
I think you can still assert_le!(usize::from_str(expect["8"]).unwrap(), size_of::<...>())

Reply:
Interesting. This works:
```
assert_eq!(
    usize::from_str_radix(expect!["1"].data(), 10).unwrap(),
    size_of::<bool>()
);
```
I need `assert_eq` here because here I'm trying to get the exact number from `size_of` for visualization. The section below (the various `assert_mem_size_equals_declared_size` calls) are the actual assertions. There we might consider switching `eq` with `le` but I think having a more strict bound is helpful unless it turns out to be too much maintenance burden. 
Also `assert_le!` is not an std macro, it is available from a 3rd party lib e.g. https://docs.rs/more-asserts/latest/more_asserts/macro.assert_le.html

Overall the above expression is more verbose than the original so I'm leaning towards keeping the original. What do you think?

Reply:
It's fine, I'm not saying the order of the instruction is wrong, I was just suggesting that you could make an assertion of a bound rather than equality in the cases where you are avoiding measuring size altogether (you have a comment there saying you don't rely on size measurement in some cases)